### PR TITLE
manage aws addons on eks cluster

### PR DESCRIPTION
**Pull request by orong-pp**

### CODE REVIEW

File: aws-eks-addon/coredns.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+resource "aws_eks_addon" "coredns" {
+  cluster_name = eks_properties.name
+  addon_name = "coredns"
+  addon_version = addon_properties.coredns_version
+  resolve_conflicts_on_update = "PRESERVE"
+  
+  node_group_name = [
```
Comment (orong-pp) [inline aws-eks-addon/coredns.tf]: I didn't see an attribute `node_group_name` in the aws_eks_addon docs
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_addon

File: aws-eks-addon/variables.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+variable "eks_properties" {}
+variable "addon_properties" { type = object({
+        coredns_version = number
+        kube_proxy_version = number
+        vpc_cni_version = number
+})}
```
Comment (orong-pp) [inline aws-eks-addon/variables.tf]: you can change addon properties to 

```suggestion
variable "addon_properties" { type = map(string) }
```

the keys will be the addon name and the values will be the addon version

File: aws-eks-addon/vpc_cni.tf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+resource "aws_eks_addon" "vpc_cni" {
+  cluster_name = eks_properties.name
+  addon_name = "vpc-cni"
+  addon_version = addon_properties.vpc_cni_version
+  resolve_conflicts_on_update = "PRESERVE"
+  
+  node_group_name = [
+    eks_properties.node_security_group_name
+  ]
+}
```
Comment (orong-pp) [inline aws-eks-addon/vpc_cni.tf]: you can change the addon to be one file in the root directory with a for each on the addon properties list

```suggestion
resource "aws_eks_addon" "eks_addon" {
    for_each = var.addon_properties

  cluster_name = eks_properties.name
  addon_name = each.key
  addon_version = each.value
  resolve_conflicts_on_update = "PRESERVE"
}
```

File: aws-eks-addon/coredns.tf (hunk 1 above)
Comment (litalyo) [reply aws-eks-addon/coredns.tf]: fixed

File: aws-eks-addon/variables.tf (hunk 2 above)
Comment (litalyo) [reply aws-eks-addon/variables.tf]: fixed

File: aws-eks-addon/vpc_cni.tf (hunk 3 above)
Comment (litalyo) [reply aws-eks-addon/vpc_cni.tf]: fixed

**Pull request merged**